### PR TITLE
feat(api): introduce selectors.register method

### LINKED ISSUE #686: Expose API for pluggable selectors
Playwright supports pluggable selector engines, let's expose it to the clients.

```js
const { chromium, webkit, firefox, selectors } = require('playwright');
const { a11y } = require('playwright-a11y-selector');
selectors.register(a11y);
```

### PR DESCRIPTION
**Pull request by dgozman**
Fixes #686.

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -3196,6 +3203,60 @@ Contains the status text of the response (e.g. usually an "OK" for a success).
...
+const { selectors, firefox } = require('playwright');  // Or 'chromium' or 'webkit'.
+
+(async () => {
+  const createTagSelector = () => ({
+    // Selectors will be prefixed with "tag=".
+    name: 'tag',
+
+    // Creates a selector which matches given target when queried at the root.
+    create(root, target) {
+      return target.nodeName;
```
Comment (pavelfeldman) [inline docs/api.md]: tagName for clarity

File: docs/api.md [hunk 2]
```diff
@@ -3196,6 +3203,60 @@ Contains the status text of the response (e.g. usually an "OK" for a success).
...
+    query(root, selector) {
+      return root.querySelector(selector);
+    },
+
+    // Returns all elements matching given selector in the root's subtree.
+    queryAll(root, selector) {
+      return Array.from(root.querySelectorAll(selector));
+    }
+  });
+  // Construct an expression which evaluates to our selector instance.
```
Comment (pavelfeldman) [inline docs/api.md]: Blank line

**Review by pavelfeldman — APPROVED** [review]

File: docs/api.md (hunk 1 above)
Comment (dgozman) [reply docs/api.md]: Done.

File: docs/api.md (hunk 2 above)
Comment (dgozman) [reply docs/api.md]: Done.

**Pull request merged**